addons/io_hubs_addon/components/definitions/ammo_shape.py

```diff
@@ -0,0 +1,89 @@
+               ("mesh", "Mesh Collider", "A shape made of the actual vertecies of the object. This can be expensive for large meshes.")],
+        default="hull")
+
+    # TODO Add conditional UI to show only the required properties per type
+    fit: EnumProperty(
+        name="Shape Fitting Mode",
+        description="Shape fitting mode",
+        items=[("all", "Automatic fit all", "Automatically match the shape to fit the object's vertecies"),
+               ("manual", "Manual fit", "Use the manually specified dimensions to define the shape, ignoring the object's vertecies")],
+        default="all")
+
+    halfExtents: FloatVectorProperty(
+        name="Half Extents",
+        description="Half dimensions of the collider. (Only used when fit is set to \"manual\" and type is set ot \"box\").",
+        unit='LENGTH',
+        subtype="XYZ",
+        default=(0.5, 0.5, 0.5))
+
+    minHalfExtent: FloatProperty(
+        name="Min Half Extent",
+        description="The minimum size to use when automatically generating half extents. (Only used when fit is set to \"all\" and type is set ot \"box\")",
+        unit="LENGTH",
+        default=0.0)
+
+    maxHalfExtent: FloatProperty(
+        name="Max Half Extent",
+        description="The maximum size to use when automatically generating half extents. (Only used when fit is set to \"all\" and type is set ot \"box\")",
+        unit="LENGTH",
+        default=1000.0)
+
+    sphereRadius: FloatProperty(
+        name="Sphere Radius",
+        description="Radius of the sphere collider. (Only used when fit is set to \"manual\" and type is set ot \"sphere\")",
+        unit="LENGTH",
+        default=0.5)
+
+    offset: FloatVectorProperty(
+        name="Offset", description="An offset to apply to the collider relative to the object's origin.",
+        unit='LENGTH',
+        subtype="XYZ",
+        default=(0.0, 0.0, 0.0))
+
+    includeInvisible: BoolProperty(
+        name="Include Invisible",
+        description="Include invisible objects when generating a collider. (Only used if \"fit\" is set to \"all\")",
+        default=False)
+
+    def draw(self, context, layout):
+        super().draw(context, layout)
+
+        parents = [context.object]
+        while parents:
+            parent = parents.pop()
+            if parent.scale != V_S1:
+                col = layout.column()
+                col.alert = True
+                col.label(
+                    text="The ammo-shape object, and it's parents, scale needs to be [1,1,1]", icon='ERROR')
+
+                break
+
+            if parent.parent:
+                parents.insert(0, parent.parent)
+
+            if parent.parent_bone:
+                parents.insert(0, parent.parent_bone)
```

Comment:
Sorry, I made a mistake with this.
This should fix it:
```suggestion
            if hasattr(parent, 'parent_bone') and parent.parent_bone:
                parents.insert(0, parent.parent.pose.bones[parent.parent_bone])
```

Reply:
Updated